metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala

```diff
@@ -89,4 +87,11 @@ class SemanticdbIndexer(
       }
     }
   }
+
+  private def onChange(path: AbsolutePath, docs: TextDocuments): Unit = {
+    referenceProvider.onChange(docs, path)
+    implementationProvider.onChange(docs, path)
+    implicitDecorator.onChange(docs, path)
+    testSuitesProvider.onChange(docs, path)
```

Comment:
> Shouldn't it also be called in onDelete method?

I can add that to be consistent.

> btw. I think it would be good to refactor make some command trait for providers.

I thought of something similar, but Tomek already sent me "`+942 −353  😞`" after I asked him for review hah :D